duckduckgo/Android
# Improve android back navigation

Task/Issue URL: https://app.asana.com/0/361428290920652/1117886846776279
Tech Design URL: https://app.asana.com/0/361428290920652/1118398381563523 and https://app.asana.com/0/361428290920652/1118398381563522

**Description**:
Improves three aspects of android back navigation:
- Pressing back after launching from another app takes user back to the previous application (rather than via home screen)
- When pressing back to get to the home screen, users can now press forward again to return to webpage they were viewing
- Pressing back on a https-upgraded site navigates two steps to avoid sending user back to the http site which just upgrades again

**Steps to test this PR**:
MIGRATION TEST
1. Install the develop version of the app.
1. Open some tabs
1. Install this version and make sure nothing goes "boom"

FORWARD TEST
1. Go to a webpage
1. Press the back button to go back to the home page
1. Press the forward button to go back to the webpage

SHARE TEST
1. Share a link from twitter to our app
1. Now from our app, press back
1. This should return you to twitter (without going via the home screen)

HTTPS Test
1. Navigate to http://duckduckgo.com
1. Now navigate to http://facebook.com
1. Press back, this should take you to https://duckduckgo.com
1. Press back again, this should take you to the native home screen
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
